CorsixTH/Lua/dialogs/resizables/file_browser.lua

```diff
@@ -197,23 +197,38 @@ function UIFileBrowser:UIFileBrowser(ui, mode, title, vertical_size, root, show_
   -- Initialize the tree control
   self.control = FilteredTreeControl(root, 5, 35, h_size - 10, vertical_size, self.col_bg, self.col_scrollbar, true, show_dates)
     :setSelectCallback(--[[persistable:file_browser_select_callback]] function(node)
-      if node.is_valid_file and (lfs.attributes(node.path, "mode") ~= "directory") then
-        self:choiceMade(node.path)
-      end
+      self:checkChoice(node)
     end)
   self:addWindow(self.control)
 
-  -- Create the back button.
-  self:addBevelPanel((h_size - 160) / 2, 340, 160, 30, self.col_bg):setLabel(_S.menu_list_window.back)
-    :makeButton(0, 0, 160, 40, nil, self.buttonBack):setTooltip(_S.tooltip.menu_list_window.back)
+  -- Create the back and ok buttons.
+  local button_size = 135
+  local indent = math.floor((h_size - (2*button_size))/3)
+  self:addBevelPanel(indent, 340, button_size, 30, self.col_bg):setLabel(_S.menu_list_window.back)
+    :makeButton(0, 0, button_size, 40, nil, self.buttonBack):setTooltip(_S.tooltip.menu_list_window.back)
+
+  self:addBevelPanel(h_size - button_size - indent, 340, button_size, 30,
+  self.col_bg):setLabel(_S.menu_list_window.ok)
+    :makeButton(0, 0, button_size, 40, nil, (--[[persistable:filebrowser_ok_callback]] function()
+      if (self.new_savegame_textbox and self.new_savegame_textbox.text ~= "") or (self.new_map_textbox and self.new_map_textbox.text ~= "") then
```

Comment:
Ideally there would be a parameter of file_browser for `save` mode which could be checked. The `confirmName` function definition isn't in this file, so the textboxes don't otherwise leak. Empty name should (and is) handled inside the function. I don't want to see controls for specific dialogs referenced in this file.

You might do `if self.confirmName then self:confirmName() elseif ...`